### PR TITLE
Improve standalone inference script and new metrics scripts

### PR DESCRIPTION
**Pull request by mintar**
Details can be found in the commit messages of the separate commits.

### CODE REVIEW

File: scripts/metrics/render_json.py [hunk 1]
```diff
@@ -162,7 +170,15 @@ def create_obj(
...
 visii.set_dome_light_intensity(1)
-visii.set_dome_light_color(visii.vec3(1,1,1),0)
+
+try:
+    visii.set_dome_light_color(visii.vec3(1, 1, 1), 0)
+except TypeError:
+    # Support for alpha transparent backgrounds was added in nvisii ef1880aa,
+    # but as of 2022-11-03, the latest released version (1.1) does not include
+    # that change yet.
+    print("WARNING! Your version of NVISII does not support alpha transparent backgrounds yet; --contour will not work properly.")
```
Comment (TontonTremblay) [inline scripts/metrics/render_json.py:180]: Good catch I like it. :P We are working on making the alpha changes on pip, but things get in the way.

File: scripts/train2/inference.py [hunk 2]
```diff
@@ -310,10 +329,6 @@ def rotate_vector(vector, quaternion):
 
     # create the output folder
     print (f"output is located in {opt.outf}")
-    try:
-        shutil.rmtree(f"{opt.outf}")
-    except:
-        pass
 
     try:
         os.makedirs(f"{opt.outf}")
```
Comment (TontonTremblay) [inline scripts/train2/inference.py:334]: yeah much better

File: scripts/train2/inference/detector.py [hunk 3]
```diff
@@ -748,7 +746,7 @@ def find_objects(vertex2, aff, config, numvertex=8, run_sampling=False, num_samp
                             best_angle = dist_angle
                             best_dist = dist_point
 
-                    if i_best is -1:
```
Comment (TontonTremblay) [inline scripts/train2/inference/detector.py:751]: you dont like `is`?

**Review by TontonTremblay — COMMENTED** [review]
Great commit, thank you for the update, it will make the code more stable.

File: scripts/train2/inference/detector.py (hunk 3 above)
Comment (mintar) [reply scripts/train2/inference/detector.py:751]: Python doesn't like it in this case. Python 3.8.10 threw a SyntaxWarning (something like `'is' used with literal`).

**Comment by mintar** [comment]
Thanks for mething so quickly, and thanks for the code, very useful as always!

**Pull request merged**